File: src/main/java/org/zwobble/mammoth/internal/util/Base64Encoding.java
package org.zwobble.mammoth.internal.util;

import java.io.IOException;
import java.io.InputStream;
import java.util.Base64;

import static org.zwobble.mammoth.internal.util.Streams.toByteArray;

public class Base64Encoding {
    public static String streamToBase64(SupplierWithException<InputStream, IOException> open) throws IOException {
        try (InputStream stream = open.get()) {
            return Base64Encoding.streamToBase64(stream);
        }
    }

    public static String streamToBase64(InputStream stream) throws IOException {
        return Base64.getEncoder().encodeToString(toByteArray(stream));
    }
}

File: dotnet/Mammoth/Couscous/mammoth/internal/util/Base64Encoding.cs
using Mammoth.Couscous.java.lang;
using Mammoth.Couscous.java.io;

namespace Mammoth.Couscous.org.zwobble.mammoth.@internal.util {
    internal static class Base64Encoding {
        internal static string streamToBase64(SupplierWithException<InputStream, IOException> streamSupplier) {
            using (var stream = streamSupplier.get().Stream) {
                return streamToBase64(stream);
            }
        }
        
        internal static string streamToBase64(System.IO.Stream stream) {
            var memoryStream = new System.IO.MemoryStream();
            stream.CopyTo(memoryStream);
            return System.Convert.ToBase64String(memoryStream.ToArray());
        }
    }
}
